integration_tests/src/main/python/hybrid_parquet_test.py

```diff
@@ -290,6 +290,413 @@ def test_hybrid_parquet_filter_pushdown_timestamp(spark_tmp_path):
+    "(map_values(map(int_not_null, str1)) == array('b'))",
+    "(isnotnull(map_zip_with(map(int1, str1, int2, str2), map(int3, str3), (k, v1, v2) -> if(k == int3, v2, v1))))",
+    "(next_day(date1, 'monday') == date1)",
+    "(overlay(str1, 'a', 1) == 'hallo')",
+    "(isnotnull(str_to_map(str1)))",
+    "(substring_index(str1, 'e', 2) == 'he')",
+    pytest.param("(to_unix_timestamp(timestamp) == 1717171717)", marks=pytest.mark.xfail(reason='timestamp filter not supported by gluten')),
+    "(unhex('68656c6c6f') == 'hello')",
+    # Ansi false only:
+    # Multiply, Add, Subtract, Divide, Abs, Pmod
+    "(int1 * int2 == 24)",
+    "(int1 + int2 == 5)",
+    "(int1 - int2 == 1)",
+    "(int1 / int2 == 2)",
+    "(abs(int1) == 1)",
+    "(pmod(double1, double2) == 1)",
+]
+
+@pytest.mark.skipif(is_databricks_runtime(), reason="Hybrid feature does not support Databricks currently")
+@pytest.mark.skipif(not is_hybrid_backend_loaded(), reason="HybridScan specialized tests")
+@hybrid_test
+@pytest.mark.parametrize('condition', condition_list, ids=idfn)
+def test_hybrid_parquet_filter_pushdown_more_exprs(spark_tmp_path, condition):
+    data_path = spark_tmp_path + '/PARQUET_DATA'
+    genlist = [('bool1', BooleanGen()),
+               ('bool2', BooleanGen()),
+               ('double1', DoubleGen()),
+               ('double2', DoubleGen()),
+               ('double3', DoubleGen()),
+               ('int1', IntegerGen()),
+               ('int2', IntegerGen()),
+               ('int3', IntegerGen()),
+               ('int4', IntegerGen()),
+               ('int_not_null', IntegerGen(nullable=False)),
+               ('date1', DateGen()),
+               ('date2', DateGen()),
+               ('timestamp', TimestampGen(start=datetime(1900, 1, 1, tzinfo=timezone.utc))),
+               ('array_int_1', ArrayGen(IntegerGen())),
+               ('array_int_2', ArrayGen(IntegerGen())),
+               ('array_array_int_1', ArrayGen(ArrayGen(IntegerGen(),max_length=3))),
+               ('array_str_1', ArrayGen(StringGen())),
+               ('str1', StringGen()),
+               ('str2', StringGen()),
+               ('str3', StringGen()),
+               ('digitstr', StringGen(pattern='[0-9]{1,5}')),
+               ('long', LongGen()),
+               ('comma_separated_str', StringGen(pattern='([0-9],){1,5}')),
+               ('json_str', StringGen(pattern=r'\{"a": "[a-z]{1,3}"\}')),
+    ]
+    with_cpu_session(
+        lambda spark: gen_df(spark, genlist).write.parquet(data_path),
+        conf=rebase_write_corrected_conf)
+
+    conf = {
+        'spark.sql.sources.useV1SourceList': 'parquet',
+        'spark.rapids.sql.hybrid.parquet.enabled': 'true',
+        'spark.rapids.sql.parquet.pushDownFiltersToHybrid': 'CPU',
+        'spark.rapids.sql.exec.FilterExec': False,
+        'spark.sql.ansi.enabled': 'false',
+        'spark.rapids.sql.hybrid.whitelistExprs': 'Cast'
+    }
+
+    assert_gpu_and_cpu_are_equal_collect(
```

Comment:
Please use `assert_cpu_and_gpu_are_equal_collect_with_capture` instead of `assert_gpu_and_cpu_are_equal_collect` to ensure HybridScan is indeedly enabled.

Reply:
done.